### ISSUE TITLE
the hidden field on the technology prototype isn't respected, which means showing internal mod research

### ISSUE THREAD

**Issue by ahicks92** [Factorio-Access/FactorioAccess#162]
## **1. Describe the bug**

You can see the hidden researches driving QOL: https://mods.factorio.com/mod/qol_research

## **2. Steps to Reproduce**

Install QOL. Check list of technologies. Notice qolinternal.

## **3. Expected correct behavior**

Not seeing that.

## **4. Mod version**

Tip of next-update.

**Comment by ahicks92** [comment]
I did try to fix this but couldn't figure out which of several places needed checks.

**Comment by LevFendi** [comment]
This is similar to issue #150 for recipes added by other mods. For both cases it should be easy to check the "hidden" property of the respective prototype. I will look into it, thank you.

**Comment by ahicks92** [comment]
Yeah I figure it's easy for you but I was finding stuff that was comparing to magic numbers and some sort of cache and was like....

Sorry I couldn't fix it.  Because this should be the sort of thing I can fix, in theory.

**Comment by ahicks92** [comment]
Nominally handled but vanilla doesn't have hidden tech to test on.

Whenever someone breaks it we can deal with it then.  Right now we cna't inject that the test save should enable prototypes for testing so we can't add in some ourselves.
